### ISSUE TITLE
Getting the correct answer via recursion, but the system won't accept

### ISSUE THREAD

**Issue by zmasinelli** [FreeCodeCamp/FreeCodeCamp#5457]
Challenge [Bonfire: Roman Numeral Converter](http://www.freecodecamp.com/challenges/bonfire-roman-numeral-converter#?solution=%20var%20arr%20%3D%20%5B%5D%3B%0A%0Afunction%20convert(num)%20%7B%0A%20%20%20%20%0A%20%20%0A%20%20if(num%20%3E%201000)%7B%0A%20%20%20%20arr.push('M')%3B%0A%20%20%20%20num%20-%3D%201000%3B%0A%20%20%20%20return%20convert(num)%3B%0A%20%20%7D%0A%20%20%0A%20%20if(num%20%3E%3D%20500)%7B%0A%20%20%20%20arr.push('D')%3B%0A%20%20%20%20num%20-%3D500%3B%0A%20%20%20%20return%20convert(num)%3B%0A%20%20%7D%0A%20%20%0A%20%20if(num%20%3E%3D%20100)%7B%0A%20%20%20%20arr.push('C')%3B%0A%20%20%20%20num%20-%3D100%3B%0A%20%20%20%20return%20convert(num)%3B%0A%20%20%7D%0A%20%20%0A%20%20if(num%20%3E%3D%2050)%7B%0A%20%20%20%20arr.push('L')%3B%0A%20%20%20%20num%20-%3D50%3B%0A%20%20%20%20return%20convert(num)%3B%0A%20%20%7D%0A%20%20%0A%20%20if(num%20%3E%3D%2010)%7B%0A%20%20%20%20arr.push('X')%3B%0A%20%20%20%20num%20-%3D10%3B%0A%20%20%20%20return%20convert(num)%3B%0A%20%20%7D%0A%20%20%0A%20%20if(num%20%3E%3D%205)%7B%0A%20%20%20%20arr.push('V')%3B%0A%20%20%20%20num%20-%3D5%3B%0A%20%20%20%20return%20convert(num)%3B%0A%20%20%7D%0A%20%20%0A%20%20if(num%20%3E%3D%201)%7B%0A%20%20%20%20arr.push('I')%3B%0A%20%20%20%20num%20-%3D1%3B%0A%20%20%20%20return%20convert(num)%3B%0A%20%20%7D%0A%20%0A%20%20var%20total%20%3D%20arr.join('')%3B%0A%20%20%0A%20return%20arr%3B%0A%7D%0A%0Aconvert(5)%3B%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.106 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

I run it and get the solution (at the very least, convert(5) results in 'V'). I've tried returning both a string and an array, and it accepts neither. Is there a rule against using recursion that I don't know of?

Thanks!

My code:
```javascript
 var arr = [];

function convert(num) {
    
  
  if(num > 1000){
    arr.push('M');
    num -= 1000;
    return convert(num);
  }
  
  if(num >= 500){
    arr.push('D');
    num -=500;
    return convert(num);
  }
  
  if(num >= 100){
    arr.push('C');
    num -=100;
    return convert(num);
  }
  
  if(num >= 50){
    arr.push('L');
    num -=50;
    return convert(num);
  }
  
  if(num >= 10){
    arr.push('X');
    num -=10;
    return convert(num);
  }
  
  if(num >= 5){
    arr.push('V');
    num -=5;
    return convert(num);
  }
  
  if(num >= 1){
    arr.push('I');
    num -=1;
    return convert(num);
  }
 
  var total = arr.join('');
  
 return arr;
}

convert(5);

```

**Comment by ltegman** [comment]
You are initializing `arr` globally, which means you will not start with an empty array the second time the function is run and will return a wrong result. In the future please visit the [Help Chat](https://gitter.im/FreeCodeCamp/Help) to confirm your issue is a bug.

Thanks and happy coding!

**Comment by zmasinelli** [comment]
Even with the variable initialized within the loop it wasn't working - I had moved it outside as a test to see if the system would accept it that way (I was trying anything I could think of). I forgot to change that back before posting here.

Eventually I tried through a different browser with the same code and it worked, so it must have just been a weird cache issue. In any case, I got past the module.

Thanks for your input! Much appreciated :)
